CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.20)
+project(libCEED LANGUAGES C CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
+
+if (CEED_ENABLE_CUDA)
```

Comment:
Are these undocumented `cmake -DCEED_ENABLE_CUDA=1`? It fails for me on a vanilla Linux system
```console
$ cmake . -DCEED_ENABLE_CUDA=1
-- The CUDA compiler identification is NVIDIA 12.4.99
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /opt/cuda/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Configuring done (1.7s)
-- Generating done (0.0s)
-- Build files have been written to: /home/jed/src/libceed/build-cmake
$ make
[  7%] Building C object CMakeFiles/CEED.dir/interface/ceed-jit-source-root-default.c.o
[  7%] Building C object CMakeFiles/CEED.dir/interface/ceed-basis.c.o
[  7%] Building C object CMakeFiles/CEED.dir/interface/ceed-cuda.c.o
[  7%] Building C object CMakeFiles/CEED.dir/interface/ceed-elemrestriction.c.o
[  7%] Building C object CMakeFiles/CEED.dir/interface/ceed-fortran.c.o
[  8%] Building C object CMakeFiles/CEED.dir/interface/ceed-jit-source-root-install.c.o
[ 11%] Building C object CMakeFiles/CEED.dir/interface/ceed-operator.c.o
[ 11%] Building C object CMakeFiles/CEED.dir/interface/ceed-jit-tools.c.o
In file included from /home/jed/src/libceed/interface/ceed-cuda.c:11:
/home/jed/src/libceed/include/ceed/cuda.h:12:10: fatal error: cuda.h: No such file or directory
   12 | #include <cuda.h>
      |          ^~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/CEED.dir/build.make:90: CMakeFiles/CEED.dir/interface/ceed-cuda.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 14%] Building C object CMakeFiles/CEED.dir/interface/ceed-preconditioning.c.o
[ 14%] Building C object CMakeFiles/CEED.dir/interface/ceed-qfunction-register.c.o
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/CEED.dir/all] Error 2
```

Reply:
I wrote this PR one morning on my laptop, so the initial CUDA bits were sort of just a placeholder. I'm revisiting now on a CUDA workstation, and it seems to be building successfully after a few small changes.